### PR TITLE
add special treatment for duplicate errors when importing calendars

### LINKED ISSUE #589: Import only events whose UID does not already exist
### Steps to reproduce
1. Import ics.
2. Import same ics again.

### Expected behaviour
It will be only imported events, whose UID does not already exist in calender for the user.

### Actual behaviour
Every event is imported regardless if it already exist or not. Therefore it exists events with the same UID in the same calender, which is not allowed by CalDAV specification: https://tools.ietf.org/html/rfc4791.

> 
>  The UID property value of the calendar components contained in a
>    calendar object resource MUST be unique in the scope of the calendar
>    collection in which they are stored.
> 
>    Calendar components in a calendar collection that have different UID
>    property values MUST be stored in separate calendar object resources.
> 
>    Calendar components with the same UID property value, in a given
>    calendar collection, MUST be contained in the same calendar object
>    resource.  This ensures that all components in a recurrence "set" are
>    contained in the same calendar object resource.  It is possible for a
>    calendar object resource to just contain components that represent
>    "overridden" instances (ones that modify the behavior of a regular
>    instance, and thus include a RECURRENCE-ID property) without also
>    including the "master" recurring component (the one that defines the
>    recurrence "set" and does not contain any RECURRENCE-ID property).

I already found #53, which was closed earlier this year. However I confirm that it makes no sense to import already existing events. Therefore I would not ask the user about replacing / adding the event. Instead, the item to be imported should be ignored if another one already exists with the same UID in the calendar.

Independent of the CalDAV specification, here is an use case:

User deletes an event and needs to restore a backup. The backup does not contain all events of the current calendar but the event, which was deleted by accident. Therefore I cannot delete the whole calendar (because I would delete events, which are not in the backup), but I also cannot import the ics because of the duplicated events.

### PR DESCRIPTION
**Pull request by georgehrke**
fixes #589

### CODE REVIEW

File: js/app/filters/importerrorfilter.js [hunk 1]
```diff
@@ -29,17 +29,29 @@ app.filter('importErrorFilter', function () {
...
-		switch(file.errors) {
-			case 0:
-				return t('calendar', 'Successfully imported');
-
-			case 1:
+		if (file.errors === 0) {
+			return t('calendar', 'Successfully imported');
+		} else if(file.errors === 1) {
+			if (file.duplicates === 1) {
+				return t('calendar', 'Partially imported, skipped 1 duplicate');
```
Comment (tcitworld) [inline js/app/filters/importerrorfilter.js]: 1 -> one

File: js/app/filters/importerrorfilter.js [hunk 2]
```diff
@@ -29,17 +29,29 @@ app.filter('importErrorFilter', function () {
 		}
 
 		//TODO - use n instead of t to use proper plurals in all translations
```
Comment (tcitworld) [inline js/app/filters/importerrorfilter.js]: Maybe a good time to handle this ?

File: js/app/filters/importerrorfilter.js [hunk 3]
```diff
@@ -29,17 +29,29 @@ app.filter('importErrorFilter', function () {
...
-		switch(file.errors) {
-			case 0:
-				return t('calendar', 'Successfully imported');
-
-			case 1:
+		if (file.errors === 0) {
+			return t('calendar', 'Successfully imported');
```
Comment (tcitworld) [inline js/app/filters/importerrorfilter.js]: Is there a way we could bring in the number of imported events here ?

Comment (georgehrke) [reply js/app/filters/importerrorfilter.js]: Good call, will add it

File: js/app/filters/importerrorfilter.js (hunk 1 above)
Comment (georgehrke) [reply js/app/filters/importerrorfilter.js]: I don't think we can have both, `n` and `one`

**Comment by georgehrke** [comment]
@tcitworld I updated the strings, now it's more verbose

**Pull request merged**